main/src/ca/uwaterloo/flix/api/CrashHandler.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2022 Magnus Madsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.api
+
+import java.io.{IOException, PrintWriter, StringWriter}
+import java.nio.file.{Files, Path}
+
+object CrashHandler {
+
+  /**
+    * Creates a crash report for the given exception `ex`.
+    */
+  def handleCrash(ex: Throwable)(implicit flix: Flix): Unit = {
+    // Get the report.
+    val report = getCrashReport(ex)
+
+    // Print it.
+    println(report)
+
+    // Write it to a file.
+    getNextAvailableLogFile() match {
+      case None => // Nop
+      case Some(path) =>
+        try {
+          Files.writeString(path, report)
+        } catch {
+          case ex: IOException =>
+            println(s"Unable to write crash report to: '$path'.")
+        }
+    }
+  }
+
+  /**
+    * Returns an error message with auxiliary information for the given exception `ex`.
+    */
+  private def getCrashReport(ex: Throwable)(implicit flix: Flix): String = {
```

Comment:
We could also generate a "create issue" link with a pre-filled form. In the CLI it would probably be annoying to have a huge URL, but maybe the LSP could use it?

Reply:
Yes, but this part has to work in general. Also, there is some value in the programmer having to do it themselves-- we won't be accused of having bad telemetry. The message makes clear that you programmer can decide to take action.